### PR TITLE
Feat/add partner county limitations

### PR DESCRIPTION
**Pull request by phattya**
# Pull Request Template

## Issue Overview

This PR addresses #issue

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description
Doorway partners are only allowed to add listings that do not overlap with other current SF Bay Area affordable housing portals. We're limiting the counties that are available to be added via the County drop down box and adding messaging & links to the other portals. 

A subsequent PR will change the seed data to counties represented in the Bay Area jurisdiction (ie not San Mateo county, Alameda county, San Francisco county, and City of San Jose) to ensure that all test data will render correctly in the Partners portal.

## How Can This Be Tested/Reviewed?
Run Partners portal and check that the County field is limited in the Building Details section. Double check the links provided for the other Partner portals work correctly.

image

image

## Checklist:
- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: shared-helpers/src/utilities/formKeys.ts [hunk 1]
```diff
@@ -56,13 +56,14 @@ export const stateKeys = [
 ]
 
 export const countyKeys = [
+  //Removing options due to existing BayArea Bloom sites
```
Comment (lisunshiny) [inline shared-helpers/src/utilities/formKeys.ts:59]: nit space between `//` and text (did we format this file)
could you add a TODO here to re-add once the county sites have been sunsetted?

File: sites/partners/src/components/listings/PaperListingForm/sections/BuildingDetails.tsx [hunk 2]
```diff
@@ -10,13 +10,16 @@ import {
   FieldGroup,
   ListingMap,
   LatitudeLongitude,
+  AlertNotice,
 } from "@bloom-housing/ui-components"
+import { Icon, Tooltip } from "../../../../../../../doorway-ui-components"
```
[the remark this answers is not in the document]
Comment (phattya) [reply sites/partners/src/components/listings/PaperListingForm/sections/BuildingDetails.tsx]: Thanks! I had meant to do that. Fixed!

File: sites/partners/src/components/listings/PaperListingForm/sections/BuildingDetails.tsx [hunk 3]
```diff
@@ -309,6 +312,37 @@ const BuildingDetails = ({
...
         </GridCell>
+        <GridCell span={2}>
+          <AlertNotice inverted>
+            <p className="text__small-weighted" color="bloom-color-alert-dark">
+              {t("county.goToOtherPortalsTitle")}
+              <Tooltip className="ml-0" id="county-helper" text={t("county.goToOtherPortalsHover")}>
+                <Icon size="medium" symbol="info" />
+              </Tooltip>
+            </p>
+            <li className={"list-disc list-inside"}>
```
[the remark this answers is not in the document]
Comment (phattya) [reply sites/partners/src/components/listings/PaperListingForm/sections/BuildingDetails.tsx]: fixed.

File: shared-helpers/src/utilities/formKeys.ts (hunk 1 above)
Comment (phattya) [reply shared-helpers/src/utilities/formKeys.ts]: @lisunshiny I'm not sure I 100% follow your first sentence here, but hoping I addressed both of these now.

File: sites/partners/src/components/listings/PaperListingForm/sections/BuildingDetails.tsx [hunk 4]
```diff
@@ -309,6 +312,37 @@ const BuildingDetails = ({
...
+              {t("county.goToOtherPortalsCitySanJose")}
+              <Link href="https://partners.housingbayarea.org/">
```
[the remark this answers is not in the document]
Comment (phattya) [reply sites/partners/src/components/listings/PaperListingForm/sections/BuildingDetails.tsx:325]: It's actually 2 different portals (Bloom & DAHLIA), but I get your point. I'll see what I can do here, but I think all the other items are addressed.

**Review by lisunshiny — APPROVED** [review]

**Review by ifranch — APPROVED** [review]
Tested the partners portal and it works as expected. Lgtm!

**Pull request merged**